main.tf

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+resource "google_pubsub_topic" "main" {
```

Comment:
GCP background cloud functions can be bound to a number of [event types](https://cloud.google.com/functions/docs/concepts/events-triggers), are we just targeting pubsub events for v1?

Reply:
As we discussed out of band, we will remove the hard dependency on a specific event and turn that in to a variable prior to releasing 0.1.0. This will significantly increase the flexibility of the root module. We can then provide specific event implementations as submodules for convenience, while allowing users to define their own event implementations as desired.